### DISCUSSION TITLE
feat: Ability to embed the agent dashboard inside an iframe

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#8450] [Feature Requests]
> I'm trying to embed the agent dashboard inside an iframe but I keep having issue as browser can't load this. Is there configuration option to let the admin dashboard be embedded inside an iframe?

ref: https://discord.com/channels/647412545203994635/650257251856285707/957967475377586178

**Comment by kubre** [comment]
Currently, if you try to embed the agent dashboard interface inside an Iframe you get an error. Only iframes with same-origin can embed the agent dashboard. 

Also even if you add a header inside code for Content-Security-Policy manually, the application Logs you in but you won't be redirected to the dashboard automatically.

Can there be an option to toggle on so the application can be embedded into an iframe. Something like rocket chat wherein settings you can toggle an option to allow it to be embedded or even in env variable

**Comment by sojan-official** [comment]
@kubre After talking to the team, the conclusion is that at the moment there isn't a clean way to implement this. 
We will look into ways to solve this in upcoming sprints if there is enough interest around this. 

Meanwhile, you could take a look into the Platform APIs to generate SSO URLs for accounts, which you can use to open the interface in a new tab/embedding in frames.

https://github.com/chatwoot/chatwoot/wiki/Building-on-Top-of-Chatwoot%3A-Platform-APIs

**Comment by kubre** [comment]
@sojan-official This is helpful and for now working great also
